crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -252,21 +252,38 @@ impl<'src> Parser<'src> {
         lhs
     }
 
+    /// Parses a name.
+    ///
+    /// For an invalid name, the `id` field will be an empty string and the `ctx`
+    /// field will be [`ExprContext::Invalid`].
+    ///
+    /// See: <https://docs.python.org/3/reference/expressions.html#atom-identifiers>
     pub(super) fn parse_name(&mut self) -> ast::ExprName {
         let identifier = self.parse_identifier();
```

Comment:
Nit: Should we change `parse_identifier` to return a `Result` (`Result<Identifier, Identifier>`) to make the "invalid" check more explicit and require it for all call-sites (doesn't need to be part of this PR). 

Would that allow removing `is_valid` from` Identifier`?

Reply:
Using `Result` is a good idea to enforce this from type perspective but I don't think it'll allow us to remove the `is_valid`. The reason being that certain nodes store the `Identifier` directly and not convert it to `ExprName`. For those nodes, it'll be useful to check if the identifier is valid or not.